cql3/statements/select_statement.cc

```diff
@@ -1590,7 +1590,7 @@ bool select_statement::contains_alias(const column_identifier& name) const {
 lw_shared_ptr<column_specification> select_statement::limit_receiver(bool per_partition) {
     sstring name = per_partition ? "[per_partition_limit]" : "[limit]";
     return make_lw_shared<column_specification>(keyspace(), column_family(), ::make_shared<column_identifier>(name, true),
-        int32_type);
+        long_type);
```

Comment:
I think this breaks during mixed-version operation. The client may send a value for one of the types and the server may have a prepared statement for the other type. The sizes don't match, and validation will break.

Reply:
In any case, I don't see a query such as `SELECT * FROM tab LIMIT 5000000000` happening. People want everything or just a few.